python/pyiceberg/types.py

```diff
@@ -268,6 +268,10 @@ def __init__(self, *fields: NestedField, **data: Any):
             data["fields"] = fields
         super().__init__(**data)
 
+    def by_id(self) -> Dict[int, NestedField]:
```

Comment:
In Java, we use `field(int) -> NestedField` and `field(String) -> NestedField` methods. Can we do that instead to keep a consistent API?

Reply:
Good call, updated!